pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -2287,13 +2287,11 @@ func (ds *DistSender) sendToReplicas(
 		log.Fatalf(ctx, "unknown routing policy: %s", ba.RoutingPolicy)
 	}
 
-	// Convert unknown connection classes to DEFAULT, to handle mixed-version
-	// states gracefully.
-	rpcClass := rpc.ConnectionClassOrDefault(ba.ConnectionClass)
-	// Upgrade the connection class to SYSTEM, for critical ranges.
-	rpcClass = rpc.ConnectionClassForKey(desc.RSpan().Key, rpcClass)
```

Comment:
nit: should this change be in the previous commit?

Reply:
Done.